dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/validation/LinkValidator.java

```diff
@@ -35,12 +37,86 @@ private LinkValidator() {}
+            }
+
+            if (t.getOperationRef() != null && !t.getOperationRef().isEmpty()) {
+                if (t.getOperationRef().startsWith("#")) {
+                    String[] operationRef = t.getOperationRef().split("/");
+                    String pathKey = operationRef[2].replace("~1", "/").replace("~0", "~");
+                    Paths paths = context.getModel().getPaths();
+                    Boolean isValid = true;
+
+                    if (paths.containsKey(pathKey) && paths.get(pathKey) != null) {
+                        String op = operationRef[3];
+                        switch (op) {
+                            case "get":
+                                if (paths.get(pathKey).getGET() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "put":
+                                if (paths.get(pathKey).getPUT() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "post":
+                                if (paths.get(pathKey).getPOST() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "delete":
+                                if (paths.get(pathKey).getDELETE() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "trace":
+                                if (paths.get(pathKey).getTRACE() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "head":
+                                if (paths.get(pathKey).getHEAD() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "patch":
+                                if (paths.get(pathKey).getPATCH() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                            case "options":
+                                if (paths.get(pathKey).getOPTIONS() == null) {
+                                    isValid = false;
+                                }
+                                break;
+                        }
+                    } else {
+                        isValid = false;
+                    }
+                    if (!isValid) {
+                        final String message = Tr.formatMessage(tc, "linkOperationRefInvalidOrMissing", t);
```

Comment:
passing the t.getOperationRef() value to error message would be helpful instead of passing t (which wouldn't get converted properly into a String).